### ISSUE TITLE
Support for python3

### ISSUE THREAD

**Issue by IuryAlves** [gabrielfalcao/HTTPretty#307]
Hi. I need to use HTTPretty on python 3, more specifically on [moto](https://github.com/spulec/moto).

Can someone point me in the right direction to start to give python 3 support ?

**Comment by mikedlr** [comment]
Hi luryAlves;  

I haven't seen Gabriel responding on this much recently.  I guess he's busy with other things.  

I also started having a look at this a bit.  One of the things that I have noticed is that there are some weird problems with the test cases.  One test case may pass only when other test cases are run, for example.  An example of this problem is my bug #305 

My suggestion now would be that each test case should be made completely independent of the other ones by making sure that at the end of the test case all the setup done in that test case is undone.  Perhaps even having an end of test case check that makes sure that the test cases clean up.  Once that is done, simply running all the tests on Python 3 and seeing what works and what doesn't will probably give a good idea what should be fixed.  

The other suggestion that's been given (see bug #300) is to concentrate only on the latest version of python 3.    If you start on this it's possible that I will have a chance to join in and help.
